internal/internal_test.go

```diff
@@ -165,6 +167,38 @@ func TestCheckAllow(t *testing.T) {
 	}
 }
 
+func TestCheckTraceIDInAllowedRequest(t *testing.T) {
```

Comment:
You could use something like [this](https://github.com/open-policy-agent/opa/blob/main/test/e2e/distributedtracing/distributedtracing_test.go#L148) as a reference to setup the unit test.